src/services/codefixes/importFixes.ts

```diff
@@ -803,14 +803,19 @@ function shouldUseRequire(sourceFile: SourceFile, program: Program): boolean {
         return getEmitModuleKind(compilerOptions) < ModuleKind.ES2015;
     }
 
-    // 4. Match the first other JS file in the program that's unambiguously CJS or ESM
+    // 4. In --module nodenext, assume we're not emitting JS -> JS, so use
+    //    whatever syntax Node expects based on the detected module kind
+    if (sourceFile.impliedNodeFormat === ModuleKind.CommonJS) return true;
+    if (sourceFile.impliedNodeFormat === ModuleKind.ESNext) return false;
+
+    // 5. Match the first other JS file in the program that's unambiguously CJS or ESM
     for (const otherFile of program.getSourceFiles()) {
         if (otherFile === sourceFile || !isSourceFileJS(otherFile) || program.isSourceFileFromExternalLibrary(otherFile)) continue;
         if (otherFile.commonJsModuleIndicator && !otherFile.externalModuleIndicator) return true;
         if (otherFile.externalModuleIndicator && !otherFile.commonJsModuleIndicator) return false;

```

Comment:
when checking with other file, would it be better to check their implied node format as well to determine what this file probably is supposed to be.

Reply:
Because of the new checks I added, all files in the program are guaranteed to have an `undefined` `impliedNodeFormat`. `impliedNodeFormat` is always set to one of the two options I checked for, or else it’s not used at all in the program’s module mode.

Reply:
Isnt it set by looking up package.json in ancestor directory so is it not possible to have impliedNodeFormat not set on one file vs other ?

Reply:
If no package.json is found, it defaults to CJS. (This reflect’s Node’s behavior too. The choice is binary; the field is only optional because it’s not used outside of `--module nodenext`.)